### ISSUE TITLE
SummaryGeneratingListener delived in getSummary() the double result in sum by failed and succeful tests to found tests.

### ISSUE THREAD

**Issue by mourav2222** [junit-team/junit5#3365]
![image](https://github.com/junit-team/junit5/assets/76436807/00a4debc-47d4-4d71-a71a-012d7d637ba7)

>>>> testPlanExecutionFinished...
>>>> test1() : ???
>>>> userCanSearchKeyword2() : ???
>>>> Human-readable test name : ???

Test run finished after 10309 ms
[         2 containers found      ]
[         0 containers skipped    ]
[         4 containers started    ]
[         0 containers aborted    ]
[         4 containers successful ]
[         0 containers failed     ]
[         3 tests found           ]
[         0 tests skipped         ]
[         6 tests started         ]
[         0 tests aborted         ]
[         2 tests successful      ]
[         4 tests failed          ]

BUILD SUCCESSFUL in 16s
3 actionable tasks: 2 executed, 1 up-to-date
18:35:12: Execution finished ':Tester.main()'.

dependencies {

    testImplementation 'org.junit.platform:junit-platform-launcher:1.9.2'

    testImplementation platform('org.junit:junit-bom:5.9.1')
    testImplementation 'org.junit.jupiter:junit-jupiter:5.9.2'
    testImplementation 'org.selenide:selenide-selenoid:2.3.7'

    // https://mvnrepository.com/artifact/io.qameta.allure/allure-selenide
    testImplementation 'io.qameta.allure:allure-selenide:2.22.2'
    testImplementation 'io.qameta.allure:allure-junit5:2.22.2'

    annotationProcessor('info.picocli:picocli-codegen:4.7.1')
    testImplementation 'info.picocli:picocli:4.7.1'

    testImplementation('org.apache.logging.log4j:log4j-api:2.20.0')
    testImplementation('org.apache.logging.log4j:log4j-core:2.20.0')
    testImplementation('org.apache.logging.log4j:log4j-slf4j2-impl:2.20.0')
 
}

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

package ru.heisenbug.service;

import org.junit.platform.engine.TestExecutionResult;
import org.junit.platform.engine.discovery.DiscoverySelectors;
import org.junit.platform.launcher.*;
import org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder;
import org.junit.platform.launcher.core.LauncherFactory;
import org.junit.platform.launcher.listeners.SummaryGeneratingListener;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import ru.heisenbug.config.TestLauncherConfig;

import java.io.PrintWriter;

import static org.junit.platform.engine.discovery.ClassNameFilter.includeClassNamePatterns;

public  class TestLauncherService {
    final private Launcher launcher;
    private final Logger logger = LoggerFactory.getLogger(TestLauncherService.class);

    public TestLauncherService() {

        launcher = LauncherFactory.create();
    }

    public void executeTestPlan(String browser) {

        logger.info("Browser (executeTestPlan): {}", browser);

        SummaryGeneratingListener summaryGeneratingListener = new SummaryGeneratingListener();
        launcher.registerTestExecutionListeners(summaryGeneratingListener, new TestListener());

        LauncherDiscoveryRequest request = LauncherDiscoveryRequestBuilder
                .request()
                .configurationParameter(TestLauncherConfig.BROWSER.getParameter(), browser)
                .selectors(DiscoverySelectors.selectPackage("ru.heisenbug.tests"))
                .filters(includeClassNamePatterns("ru\\.heisenbug\\.tests\\.Google.*"))
                // .filters(includeClassNamePatterns("ru\\.heisenbug\\.tests\\.XXGoogle.*"))
                // .filters(includeClassNamePatterns("ru\\.heisenbug\\.tests\\..*"))
                .build();
        launcher.execute(request, summaryGeneratingListener);

        try(PrintWriter printWriter = new PrintWriter(System.out)) {
            summaryGeneratingListener.getSummary().printTo(printWriter);
        }
    }

    static class TestListener implements TestExecutionListener {

        @Override
        public void testPlanExecutionFinished(final TestPlan testPlan) {

            System.out.println(">>>> testPlanExecutionFinished...");
            TestExecutionListener.super.testPlanExecutionFinished(testPlan);
            testPlan.getRoots().stream().forEach(testIdentifier -> {
                testPlan.getDescendants(testIdentifier).stream().filter(TestIdentifier::isTest).forEach(testIdentifier1 -> {
                    System.out.println(">>>> " + testIdentifier1.getDisplayName() + " : ???");
                });

            });
        }

        @Override
        public void executionFinished(TestIdentifier testIdentifier, TestExecutionResult testExecutionResult) {

            if(testIdentifier.isTest()) {
                if (testExecutionResult.getStatus() == TestExecutionResult.Status.FAILED)
                {
                    System.out.println(">>>> " + testIdentifier.getDisplayName() + " : FAILED");
                }
                else if (testExecutionResult.getStatus() == TestExecutionResult.Status.SUCCESSFUL)
                {
                    System.out.println(">>>> " + testIdentifier.getDisplayName() + " : SUCCESS");
                }
            }
        }
    }
}

**Comment by sormuras** [comment]
Are there any dynamic tests tests involved? https://junit.org/junit5/docs/current/user-guide/#writing-tests-dynamic-tests

If yes, then the number of non-dynamic tests -  that is the number "static" tests as discovered by the engines before the actual test run - is expected to be lower (or equal) of the number of started tests.

**Comment by mourav2222** [comment]
@sormuras No, i habe no dynamic tests, only regular <code>@Test</code> static tests.
The effect happens with double counters if i add this Listener  launcher.registerTestExecutionListeners(summaryGeneratingListener, new TestListener());

If comment out this line, all counter are correct.
I think, something runs wrong in the summaryGeneratingListener.getSummary() if the TestExecutionListener will be registered.

**Comment by sbrannen** [comment]
I've edited your comment to improve the formatting. You might want to check out this [Mastering Markdown guide](https://guides.github.com/features/mastering-markdown/) for future reference.

**Comment by sbrannen** [comment]
@mourav2222, please refrain from posting screenshots of text.

**Comment by sbrannen** [comment]
If you would like for us to assist you, please provide a minimal sample application that reproduces the problem -- preferably a project that we can download and run (such as a public Git repository or a ZIP file).

**Comment by mourav2222** [comment]
no problem, this is my project where it happene. 
https://github.com/mourav2222/tests-cmd.git. Please take a branch windows

Start configuration: 'Tester - smoketest -chrome'
![image](https://github.com/junit-team/junit5/assets/76436807/798fa0fe-b930-4926-8c61-7d5577a4bbbf)

My Result:
===============================================================
>>>> testPlanExecutionFinished...
Counted Tests: 3
>>>> 1. test1() : SUCCESS
>>>> 2. userCanSearchKeyword2() : SUCCESS
>>>> 3. Human-readable test name : FAILED
===============================================================
Found Tests: 3
Succeeded Tests: 4
Failed Tests: 2
Aborted Tests: 0

Test run finished after 16579 ms
[         2 containers found      ]
[         0 containers skipped    ]
[         4 containers started    ]
[         0 containers aborted    ]
[         4 containers successful ]
[         0 containers failed     ]
[         3 tests found           ]
[         0 tests skipped         ]
[         6 tests started         ]
[         0 tests aborted         ]
[         4 tests successful      ]
[         2 tests failed          ]

**Comment by mourav2222** [comment]
thanks to tips
